ABC_WGD 
=================
   * [Install](#install)
   * [Model](#model)
   * [Parameters](#parameters)
   * [Summary statistics](#summary-statistics)
   * [Usage](#usage)

# Install  
```
chmod +x install.sh  
  
./install.py  
```
  
# Model  
**ABC_WGD** is made to investigate various models of speciation between a diploid and a tetraploid   
Species 1 is considered as being diploid, species 2 is the tetraploid.  


![model]
Prior distributions will feed the coalescent simulator [msnsam](https://github.com/rossibarra/msnsam), and are generated by [priorgen](https://github.com/popgenomics/ABC_WGD/blob/master/priorgenwgd). Priorgenwgd is a binary file from a compiled C++ code. It requires a **bpfile** and an **argfile** to work.   
Values for all parameters are written in the file **priorfile.txt**, one line per multilocus simulations.  
    
# Summary statistics  
Summary statistics are directly computed from the [msnsam](https://github.com/rossibarra/msnsam)'s output. For each locus, [mscalc](https://github.com/popgenomics/ABC_4pop/blob/master/mscalc_4pop.py) will compute:

| Statistics         | Description                                                                 |
|:-------------------|:----------------------------------------------------------------------------|
| __bialsites__          |  number of SNPs in the alignment                                            |
| __sf XY__               |  number of fixed differences between species X and Y / locus length         |
| __sx X__                |  number of exclusively polymorphic positions in species X / locus length    |
| __ss XY__               |  number of shared biallelic positions between species X and Y/ locus length |
| __pi X__                |  Tajima’s Theta within species X                                            |
| __theta X__             |  Watterson’s Theta witin species X                                          |
| __pearson_r_pi XY__    |  correlation’s coefficient for pi over orthologs between X and Y            |
| __pearson_r_theta XY__ |  correlation’s coefficient for theta over orthologs between X and Y         |
| __Dtaj X__              |  Tajima’s D for species X                                                   |
| __div XY__              |  raw divergence Dxy measured between X and Y                                |
| __netdiv XY__           |  net divergence Da measured between X and Y                                 |
| __minDiv XY__           |  smallest divergence measured between one individual from X and one from Y  |
| __maxDiv XY__           |  highest divergence measured between one individual from X and one from Y   |
| __Gmin XY__             |  minimum divergence between one sequence from X and one from Y __minDivXY__ divided by the average __divXY__                                                             |
| __FST XY__             |  FST between X and Y compute as 1-(pi_X + pi_Z) / (2*pi_XY)                                                        |

An array of statistics corresponding to the average statistics computed over loci and their standard deviation will be returned every multilocus simulation and written in the file **ABCstat.txt**.  
  
  
# Usage  
**To run the simulations, simply use the following command:**  
```
run_ABC_polyploid.py [model] [argfile] [bpfile]  
  
Ex: run_ABC_polyploid.py diso2_auto argfile_diso2_auto.txt bpfile  
```
This example will run 10,000 multilocus simulations, with an autopolyploidization and a disomic inheritance.  
  
run_ABC_polyploid.py will simply execute the pipeline  
```
priorgen | msnsam | mscalc
```
The output files are __ABCstat.txt__ (containing the computed summary statistics) and __priorfile.txt__ (containing the parameter values used to simulate the data from which the summary statistics were calculated).  

  
Model in [diso1, diso2_auto, diso2_allo, tetra1, tetra2_auto, tetra2_allo, hetero1, hetero2_auto, hetero2_allo]  
  
   
**This package requires**:  
 [pypy](https://pypy.org) (has to be linked to the user's bin)  
 [numpy](http://www.numpy.org/)  
 [msnsam](https://github.com/rossibarra/msnsam)  
  
  
Statistical comparison between "observation" and "simulations" can be made using various R libraries ([_abc_](https://cran.r-project.org/web/packages/abc/abc.pdf), [_abcrf_](https://cran.r-project.org/web/packages/abcrf/abcrf.pdf)).  
 
 
